bin/worker.js

```diff
@@ -0,0 +1,50 @@
+"use strict"
+
+const {parentPort} = require("worker_threads")
+const o = require("../ospec")
+
+function cloneErr({name, stack, message}) {
+	return {name, stack, message}
+}
+
+parentPort.on("message", async function(path) {
+	try {
+		await import(path)
+		o.run(function(results) {
+			// JSON.stringify()/.parse() is faster than the structured copy
+			// algorithm, especially for large objects. Also, postMessage
+			// seems not to like the results as they are handed by ospec.
+			// It hangs forever unless the objects are first copied
+			// manually.
+			// We also compress the results, since most tests are expected to
+			// pass on a typical run.
+			parentPort.postMessage(JSON.stringify(
```

Comment:
See my comment earlier regarding child process IPC using JSON. That could come in handy here.